DOC Update react injector docs to actually work (#187)
silverstripe/developer-docs: en/02_Developer_Guides/15_Customising_the_Admin_Interface/07_ReactJS_Redux_and_GraphQL.md

[/hint]

#### Define the app

Finally, let's make a really simple container app which holds a header and our notes component, and inject it into the DOM using entwine.

**my-module/client/src/App.js**

```js
import React from 'react';
import { inject } from 'lib/Injector';
import Notes from './components/Notes';

const App = ({ ListComponent }) => (
  <div>
    <h3>Notes</h3>
    <Notes />
  </div>
);
```

**my-module/client/src/index.js**
```js
import ReactDOM from 'react-dom';
import React from 'react';
import { ApolloProvider } from 'react-apollo';
import Injector from 'lib/Injector';
import App from './App';

Injector.ready(() => {
  const { apolloClient, store } = window.ss;

  // Assuming you've got some element in the DOM with the id "notes-app"
  $('#notes-app').entwine({
    onmatch() {
      ReactDOM.render(
        <ApolloProvider client={apolloClient} store={store}>
          <App />
        </ApolloProvider>,
        this[0]
      )
    },

    onunmatch: function() {
      ReactDOM.unmountComponentAtNode(this[0]);
    }
  })
});
```

> The `silverstripe/admin` module was generous enough to hang its `apolloClient` and `store` objects in the global namespace to be shared by other modules. We'll make use of those, and create our own app wrapped in `<ApolloProvider />`.

Reviewer: ```suggestion
The `silverstripe/admin` module provides `apolloClient` and `store` objects in the global namespace to be shared by other modules. We'll make use of those, and create our own app wrapped in `<ApolloProvider />`.
```

Reviewer 2: Oops I thought I'd made this change but looks like I forgot. Done now.

  templateName: CREATE,
  singularName: 'Note',
  pagination: false,
  params: {

>     input: 'CreateNoteInput!', // For graphql v3 use 'AppNoteCreateInputType!'

Reviewer 2: Note: Yes, for graphql v3 you have to add a `input: 'AppNoteCreateInputType!'` parameter entry, because otherwise it wants to add it with `Input` as the parameter by default, and that will fail because the admin module overrides the field name to be in lowerCamelCase!

  },
  fields: [
    'content',
    'id'
  ],
};

export default mutation;
```

[notice]
With graphql v3, the field names in the input object have to start with capital letters, and the input _type_ can't easily be overridden - it's always the first part of your namespace, then the class name, then "CreateInputType". So assuming your model's fully qualified classname is `App\Model\Note`, the input type is `AppNoteCreateInputType`.
[/notice]

> It looks like a lot of code, but if you're familiar with Apollo mutations, this is pretty standard. The supplied `mutate()` function gets mapped to a prop -- in this case `onAdd`, which the `AddForm` component is configured to invoke. We've also supplied the `singularName` as well as the template `CREATE` for the `createNote` scaffolded mutation.

Reviewer: ```suggestion
It looks like a lot of code, but if you're familiar with Apollo mutations, this is pretty standard. The supplied `mutate()` function gets mapped to a prop — in this case `onAdd`, which the `AddForm` component is configured to invoke. We've also supplied the `singularName` as well as the template `CREATE` for the `createNote` scaffolded mutation.
```

Reviewer: m-dash

Reviewer: there seems to be multiple occurrences in the document as well

Reviewer 2: Yeah those were pre-existing (I've moved some of them but didn't make new ones) - but I'll update them now that you're pointing them out :p

Reviewer 2: done

Reviewer 3: @michalkleiner, do you have any other comments or can I merge this PR?
Thank you in advance.
